Content.Server/Lathe/LatheSystem.cs

```diff
@@ -456,6 +462,34 @@ private void OnDatabaseModified(EntityUid uid, LatheComponent component, ref Tec
             // Goobstation - Lathe message on recipes update - End
         }
 
+
+        // Goobstation - Lathe Queue Reset
+        private void OnLatheQueueResetMessage(EntityUid uid, LatheComponent component, LatheQueueResetMessage args)
+        {
+            if (component.Queue.Count > 0)
+            {
+                var allMaterials = component.Queue.SelectMany(q => q.Materials);
+                var totalMaterials = new Dictionary<string, int>();
+
+                foreach (var (mat, amount) in allMaterials)
+                {
+                    if(!totalMaterials.ContainsKey(mat)) 
+                        totalMaterials[mat] = 0;
+                    totalMaterials[mat] += amount;
+                }
+
+                if(_materialStorage.CanChangeMaterialAmount(uid, totalMaterials)) {
+                    foreach (var (mat, amount) in totalMaterials)
+                        _materialStorage.TryChangeMaterialAmount(uid, mat, amount);
+                    component.Queue.Clear();
```

Comment:
Apparently rider bitches at people for the single line foreachs, I hate it but eh.
```suggestion
                if(_materialStorage.CanChangeMaterialAmount(uid, totalMaterials))
                {
                    foreach (var (mat, amount) in totalMaterials)
                    {
                        _materialStorage.TryChangeMaterialAmount(uid, mat, amount);
                    }
                    component.Queue.Clear();